agent/agent.go

```diff
@@ -121,39 +121,24 @@ type outputUnit struct {
 }
 
 // RunSingleInput runs a single input and can be called after an agent is created
-func (a *Agent) RunSingleInput(inputConfig *models.InputConfig, plugin telegraf.Input, ctx context.Context, shouldUpdatePlugin bool) error {
+func (a *Agent) RunSingleInput(input *models.RunningInput, ctx context.Context) error {
 
 	// NOTE: we can't just use `defer a.statusMutex.Unlock()` since except for the input validation,
 	// this function only returns once the gatherLoop is done -- i.e. when the plugin is stopped.
 	// Be careful to manually unlock the statusMutex in this function.
 
 	// validating if an input plugin is already running, and therefore shouldn't be run again
 	a.statusMux.Lock()
-	_, ok := a.statusMap["input"][inputConfig.Name]
+	_, ok := a.statusMap["input"][input.Config.Name]
```

Comment:
Also small thing and not super related to this line, but can we change the name of ```statusMap```? The name doesn't really make sense. Same with the ```pluginChannels``` variable? We can do this when we change things to the uuids, we can have a map from uuid to the channel to stop each plugin etc... We don't need to separate into inputs and outputs at that point.